bin/migrate-create

```diff
@@ -46,11 +49,19 @@ function create (name) {
   // Change the working dir
   process.chdir(program.chdir)
 
-  // Load compiler
+  // Register compiler only if --compiler was specified. Also, issue deprecation warning
   if (program.compiler) {
-    registerCompiler(program.compiler)
+    deprecate('--compiler is deprecated consider using --require instead')
```

Comment:
How about just moving all of this logic into `registerCompiler` so that you don't need to duplicate it in all the files?  Then just do the `if` blocks in there, something like:

```javascript
registerCompiler(program.compiler, program.require)
```

My thought here is just that if someone in the future changes one file and forgets the others it is hard to remember what yo check.

Reply:
I agree completely.

Reply:
And before I can merge this I will need this one done as well.

Reply:
What do you mean? It is done.

Reply:
Right but your addition has more than just the part you moved into there.  My intent was to suggest you move the entire thing into that method, more like this:

```javascript
function registerCompiler (c, r) {
  if (c && r) {
    deprecate('--compiler is deprecated consider using --require instead')
    return
  }

  // Require modules
  if (r) {
    program.require.forEach(require)
  }

  if (c) {
    var compiler = c.split(':')
    var ext = compiler[0]
    var mod = compiler[1]
    if (mod[0] === '.') mod = path.join(process.cwd(), mod)
    require(mod)({
      extensions: ['.' + ext]
    })
  }
}
```

That way in all the `bin` scripts you can just call `registerCompiler(program.compiler, program.require)`

Reply:
I have pushed new changes.